src/components/application_manager/src/commands/command_request_impl.cc

```diff
@@ -605,11 +627,128 @@ bool CommandRequestImpl::HasDisallowedParams() const {
           (!removed_parameters_permissions_.undefined_params.empty()));
 }
 
+bool CommandRequestImpl::PrepareResultForMobileResponse(
+        hmi_apis::Common_Result::eType result_code,
+        HmiInterfaces::InterfaceID interface) const {
+    using namespace helpers;
+    bool result = Compare<hmi_apis::Common_Result::eType, EQ, ONE>(
+        result_code,
+        hmi_apis::Common_Result::SUCCESS,
+        hmi_apis::Common_Result::WARNINGS,
+        hmi_apis::Common_Result::WRONG_LANGUAGE,
+        hmi_apis::Common_Result::RETRY,
+        hmi_apis::Common_Result::SAVED);
+    const HmiInterfaces::InterfaceState state =
+        application_manager_.hmi_interfaces().GetInterfaceState(interface);
+    result  = result || (hmi_apis::Common_Result::UNSUPPORTED_RESOURCE == result_code &&
+                         HmiInterfaces::STATE_NOT_AVAILABLE != state);
+    return result;
```

Comment:
@pestOO  I don't see reason in this changes because, if I change this method according with your advice I will need to add "Compare" for comparing result with unsuccessful result code.